server/src/main/java/org/opensearch/action/admin/cluster/state/TransportClusterStateAction.java

```diff
@@ -125,9 +125,12 @@ protected void clusterManagerOperation(
             ? clusterState -> true
             : clusterState -> clusterState.metadata().version() >= request.waitForMetadataVersion();
 
+        // action will be executed on local node, if either the request is local only (or) the local node has the same cluster-state as
+        // ClusterManager
         final Predicate<ClusterState> acceptableClusterStateOrNotMasterPredicate = request.local()
-            ? acceptableClusterStatePredicate
-            : acceptableClusterStatePredicate.or(clusterState -> clusterState.nodes().isLocalNodeElectedClusterManager() == false);
+            || !state.nodes().isLocalNodeElectedClusterManager()
```

Comment:
why did we change this?

Reply:
The predicate is used to assert & make decision (on clusterstate change) based on whether request is served by datanode / cluster manager. It is used in code path (138-168)  when `waitForMetadataVersion` is set in request.

Earlier the assumption was that the request is always executed on ClusterManager unless `request.local` is set to true. Now with term check, the request can be executed on datanode if it has latest cluster-state. Hence the change is to include addiitional condition to check if the request is served by datanode.

Reply:
I think i am getting confused, the original predicate when request.local == false,
``` acceptableClusterStatePredicate.or(clusterState -> clusterState.nodes().isLocalNodeElectedClusterManager() == false);```
what is point of this or condition predicate, knowing that it is will always clusterState.nodes().isLocalNodeElectedClusterManager() be true unless, it master switched but that transport action takes care of it.

Reply:
If the master switches, it does not want to continue waiting for the condition on the old master. But rather gives up with the exception. `cluster-manager stepped down waiting for metadata version `.